pkg/pipelineascode/match.go

```diff
@@ -185,6 +185,11 @@ func (p *PacRun) getPipelineRunsFromRepo(ctx context.Context, repo *v1alpha1.Rep
 		return nil, nil
 	}
 
+	if err := pipelineRunsWithSameName(pipelineRuns); err != nil {
```

Comment:
I think it may be better to get this inside those functions (by changing the logic to not exit too early):

https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/resolve/resolve.go#L102-L118

for pipeline and tasks, this would work as well for tkn pac resolve.